packages/api/src/context.ts

```diff
@@ -46,6 +46,27 @@ import {Client, Issuer} from 'openid-client'
 import {MailContext, MailContextOptions} from './mails/mailContext'
 import {User} from './db/user'
 import {ChallengeProvider} from './challenges/challengeProvider'
+import NodeCache from 'node-cache'
+import {logger} from './server'
+
+/**
+ * Peered article cache configuration and setup
+ */
+const ONE_DAY_IN_MS = 24 * 60 * 60 * 1000
+const peerCache = new NodeCache({
```

Comment:
`peerCache` init (including setting up event listener) should not be done in definitions file directly. Please move this logic into `createFetcher` function. In general I think whole peers/fetcher/caching concept should be moved to separate file defining those.

Reply:
@tomaszdurka: You cant if you do that you will loose the cache because it it reinitialized after for every requests.... Or did i get something wrong?

Reply:
Took another look. 

Indeed `createFetcher` is used in `createContextFromRequest`, which is indeed done each time. Still not happy with the creation of the node cache in definition file. It should be passed as `ContextOptions` into `WepublishServer` just like the `dbAdapter` and then used in `createContextFromRequest` from function argument.

You can add a function which creates this cache and then just pass it to the server constructor in your index.